components/scream/scripts/test_all_scream.py

```diff
@@ -171,7 +178,8 @@ def generate_all_baselines(self, git_baseline_head, git_head):
     ###############################################################################
         print("Generating baselines for ref {}".format(git_baseline_head))
 
-        if git_baseline_head != "HEAD":
+        if git_baseline_head != "HEAD" and not self._keep_tree:
```

Comment:
You can remove the keep_tree check here. Keep-tree can only be True if git_baseline_head is HEAD.

Reply:
True.